### PR TITLE
fix: `EvaluationAgent` metric computation bugs

### PR DESCRIPTION
**Pull request by johannaSommer**
## Description
This PR includes a few fixes for the full smashing & metrics computation pipeline:
- the `gpu_memory` metric is now accessible as a string (as input to `Task)` by setting a default for `mode`
- return formats of transformers logits is now float, fixing `inf` results for e.g. Perplexity computation
- ensuring that all metrics are returned as float, not e.g. torch Tensors or even Tensors on CUDA
- adding the functionality to raise an error for `move_to_device` when device movement fails
- adding exceptions to the inference handler registration, as some algorithms that reload the model into a different class can not be recognized as compatible with the transformers handler even though they are

## Related Issue
None.

## Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?
Verified locally with algorithms AWQ and GPTQ by computing Perplexity and GPUMemory metric.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Additional Notes
None.

### CODE REVIEW

File: src/pruna/evaluation/evaluation_agent.py [hunk 1]
```diff
@@ -88,6 +88,10 @@ def evaluate(self, model: Any) -> Dict[str, Any]:
                 )
         else:
             self.subsequent_model_results = results
+
+        for key, value in results.items():
+            if isinstance(value, torch.Tensor):
+                results[key] = value.item()
```
Comment (guennemann) [inline src/pruna/evaluation/evaluation_agent.py:94]: I assume the goal is to convert to float if the value is currently of type torch.Tensor? What if the value is of a completely different type? It might be helpful to check the types more generally and give a warning of it is neither float nor toch.Tensor.

File: src/pruna/engine/handler/handler_utils.py [hunk 2]
```diff
@@ -14,29 +14,71 @@
...
 
+HANDLER_EXCEPTIONS: dict[InferenceHandler, dict[str, set[str]]] = {
+    TransformerHandler: {
+        "quantizer": {"awq", "hqq"},
+        "compiler": {"ctranslate", "cgenerate"},
+    },
+    DiffuserHandler: {
+        "compiler": {"onediff"},
+        "quantizer": {"hqq_diffusers"},
+    },
```
Comment (guennemann) [inline src/pruna/engine/handler/handler_utils.py]: Probably better to base the exceptions not on the smash config entries, but based on the class.

Comment (begumcig) [reply src/pruna/engine/handler/handler_utils.py]: If we are sure that awq and hqq are explicitly used with I think this is also fine. But I agree that exception cases based on the class would be more explicit.

Comment (johannaSommer) [reply src/pruna/engine/handler/handler_utils.py]: thanks a lot, is adressed!

File: src/pruna/engine/handler/handler_utils.py [hunk 3]
```diff
@@ -14,29 +14,67 @@
 
 from typing import Any
 
+from pruna import SmashConfig
 from pruna.engine.handler.handler_diffuser import DiffuserHandler
 from pruna.engine.handler.handler_inference import InferenceHandler
 from pruna.engine.handler.handler_standard import StandardHandler
 from pruna.engine.handler.handler_transformer import TransformerHandler
 
+HANDLER_EXCEPTIONS: dict[InferenceHandler, dict[str, set[str]]] = {
```
Comment (guennemann) [inline src/pruna/engine/handler/handler_utils.py]: I believe you forgot to update the types because of the change. (dict[str, set[str]] is outdated.)

File: src/pruna/engine/handler/handler_utils.py [hunk 4]
```diff
@@ -14,29 +14,67 @@
...
+
     if "diffusers" in model.__module__:
         return DiffuserHandler()
     elif "transformers" in model.__module__:
         return TransformerHandler()
     else:
         return StandardHandler()
+
+
+def scan_for_exceptions(model: Any, smash_config: SmashConfig) -> InferenceHandler | None:
```
Comment (guennemann) [inline src/pruna/engine/handler/handler_utils.py]: It seems that the smash_config is no longer required. Please remove.
